content/getting-started/quick-start-my-first-webhook/step-2.md

```diff
@@ -0,0 +1,139 @@
+
+## Step by step
+
+1. Add the route in the `config/routes.yaml` file:
+
+```yaml
+app_webhook:
+    path: /webhook
+    controller: App\Controller\WebhookController::post
+```
+
+2. Create the controller in `src/Controller/WebhookController.php`
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class WebhookController
+{
+    // Thanks autowiring!
+    public function post(Request $request, LoggerInterface $logger): Response
+    {
+        // First we extract akeneo's headers.
+        $timestamp = $request->headers->get('x-akeneo-request-timestamp');
+        $signature = $request->headers->get('x-akeneo-request-signature');
+
+        // If they are null run away. The request does not come from us!
+        if (null === $timestamp || null === $signature) {
+
+            return new Response('', Response::HTTP_UNAUTHORIZED);
+        }
+
+        $body = $request->getContent();
+
+        // You should notice that I use a very bad practice here.
+        // Put that token into your .env file.
+        $secret = '3kz9y947jps0wggoc84ww80000o0ogs0ssc444w008s8kkcgw';
+
+        // This is the way we construct our signature.
+        $expectedSignature = hash_hmac('sha256', $timestamp . '.' . $body, $secret);
+
+        // You must check that the request really comes from us.
+        // To do it, we provide you the `x-akeneo-signature` header that you rebuilt before and check.
+        if (false === hash_equals($signature, $expectedSignature)) {
+
+            return new Response('', Response::HTTP_UNAUTHORIZED);
+        }
+
+        // Now let's check that the request is fresh.
+        if (time() - (int) $timestamp > 300) {
+            throw new \Exception('Request is too old (> 5min)');
+        }
+
+        // And now, do your stuff!
+        $logger->debug($body);
+
+        // You acknowledge you received the event.
+        return new Response();
+    }
+}
+```
+
+3. Launch your local webserver with `$ symfony serve --port=8000`. 
+
+![Server running](/img/getting-started/quick-start-my-first-webhook/server-running.png)
+
+::: info
+For the need of this quick start, I'm using Docker, that's why I map and expose the `8000` port. My Symfony binary is located in `bin`.
+:::
+
+4. Almost done! Go back to your PIM and configure the connection event subscriptions by replacing the Mockbin URL with your new URL. Then test it and save. Here is mine:
+
+![Server running](/img/getting-started/quick-start-my-first-webhook/event-subscription-configuration-sf-project.png)
+
+5. Finally, update a product with the API and here is the result:
+
+![Log result](/img/getting-started/quick-start-my-first-webhook/log-result.png)
+
+```json
```

Comment:
I'm not sure we should post the full response here, we'll probably forget to keep it up to date.

Reply:
As it's a cookbook, the goal is not to keep the result up to date but just to show an example. I don't think it's a problem.